test/src/test_descriptor_utils.dart

```diff
@@ -21,8 +23,32 @@ Future<d.DirectoryDescriptor> createPackage(
   List<d.Descriptor> libFiles = const [],
   List<d.Descriptor> files = const [],
 }) async {
+  final parsedYaml = yaml.loadYaml(pubspec) as Map;
+  final packageName = parsedYaml['name'];
+  final versionConstraint = (parsedYaml['environment'] as Map)['sdk'];
   final languageVersion =
-      RegExp(r"sdk: '>=(\S*)\.0(-0)? ").firstMatch(pubspec)!.group(1);
+      RegExp(r'>=(\S*)\.0(-0)? ').firstMatch(versionConstraint)!.group(1);
+  final packagesInfo = StringBuffer('''{
+  "name": "$packageName",
+  "rootUri": "../",
+  "packageUri": "lib/",
+  "languageVersion": "$languageVersion"
+}''');
+  if (parsedYaml.containsKey('dependencies')) {
+    final dependencies = parsedYaml['dependencies'] as Map;
+    for (var dep in dependencies.keys) {
+      // This only accepts 'path' deps.
+      final depConfig = dependencies[dep] as Map;
```

Comment:
Woohoo very nice

Reply:
Thanks! From memory, this was just needed to get one test passing, that was testing doc references between two test packages.